scraper.py

```diff
@@ -0,0 +1,171 @@
+        pass
+
+matter_type = "" # matter type
+matter_name = "" # matter name 
+matter_title = "" # matter title
+sponsor = "" #sponsor for the matter not done yet
+decision = "" # decsion for the matter
+v_yes = "" # people voted yes
+v_no = "" # people voted no
+
+matters = driver.find_elements(By.CSS_SELECTOR,"td[class='Title'][colspan='9']")
+for matter in matters:
+    try:
+        item = matter.find_element(By.CLASS_NAME, 'AgendaOutlineLink').text
+        if (len(item)!= 0):
+            matter_name = item[0:9]
+            matter_title = item[12:]
+            matter_type = " ".join(re.split('BY |FROM',matter_title)[0].split(' ')[1:-1])
+            print(matter_type)
+            link = driver.find_element("link text", item)
+            link.click()
+            s_matter = WebDriverWait(driver,10).until(
+                #EC.presence_of_all_elements_located((By.CLASS_NAME,"Date"))
+                #EC.presence_of_all_elements_located((By.XPATH, "/html/body/form/table/tbody/tr/td[2]/div[2]/div/div/div[4]/div[7]/div/table/tbody"))
+                EC.presence_of_all_elements_located((By.XPATH,
+                "//*[@id=\"ContentPlaceHolder1_divHistory\"]/div/table/tbody/tr"))
+            )
+            sponsor = driver.find_element(By.XPATH, "//*[@id=\"tblLegiFileInfo\"]/tbody/tr[1]/td[2]").text
+            s_rows = (len(s_matter))
+            for i in range(1, s_rows+1, 2):
+                header =  driver.find_element(By.XPATH,"//*[@id=\"ContentPlaceHolder1_divHistory\"]/div/table/tbody/tr[" + str(i) + "]")
+                date = header.find_element(By.CLASS_NAME, "Date").text
+                s_word = "Apr 18, 2022 11:00 AM"
+                if s_word in date:
+                    result =  driver.find_element(By.XPATH, "//*[@id=\"ContentPlaceHolder1_divHistory\"]/div/table/tbody/tr[" + str(i+1) + "]/td/table")
+                    decision = result.find_element(By.CLASS_NAME, "Result").text # vote result
+                    sub_sections = result.find_elements(By.XPATH, "//*[@id=\"ContentPlaceHolder1_divHistory\"]/div/table/tbody/tr[" + str(i+1) + "]/td/table/tbody/tr")
+                    for j in range(1, len(sub_sections)+1):
+                        sub_content =  driver.find_element(By.XPATH,"//*[@id=\"ContentPlaceHolder1_divHistory\"]/div/table/tbody/tr[" + str(i+1) + "]/td/table/tbody/tr[" + str(j) + "]")
+                        sub_content_role = sub_content.find_element(By.CLASS_NAME, "Role").text
+                        if "AYES" in sub_content_role:
+                            v_yes = driver.find_element(By.XPATH,"//*[@id=\"ContentPlaceHolder1_divHistory\"]/div/table/tbody/tr[" + str(i+1) + "]/td/table/tbody/tr[" + str(j) + "]/td[2]").text
+                        if "NAYS" in sub_content_role:
+                            v_no = driver.find_element(By.XPATH,"//*[@id=\"ContentPlaceHolder1_divHistory\"]/div/table/tbody/tr[" + str(i+1) + "]/td/table/tbody/tr[" + str(j) + "]/td[2]").text
```

Comment:
This looks complicated but I trust it. It would be helpful to have a comments as to what you are parsing. I think its a table of votes??

Reply:
Yes, this is the voting result, I will add comments to it to make it more clear!